### PR TITLE
feat: ✨ add `edit_resource_properties_field()`

### LINKED ISSUE #576: Create and test `edit_resource_properties_field` function
- signature: `edit_resource_properties_field(properties: dict, field: str, value: any) -> dict`
- arguments:
  - `properties`: a dictionary object containing resource properties
  - `field`: the name of the property to set
  - `value`: the value of the property to set
- returns `properties` with the property `field` set to `value`
- consider if we want to check that `field` is a valid property according to the Frictionless standard (or our own custom properties, if any)
- consider if we can describe the type of `value` more concretely

For context, see the [diagram](https://github.com/seedcase-project/seedcase-sprout/blob/main/docs/design/images/core-python-functions/create-resource-properties.svg) for `create_resource_properties`.

### PR DESCRIPTION
**Pull request by martonvago**
## Description

This adds the `edit_resource_properties_field()` function.

Closes #576 

## Reviewer Focus

This PR needs an in-depth review.

## Checklist

- [x] Added or updated tests
- [x] Tests passed locally
- [x] Linted and formatted code
- [x] Build passed locally
- [x] Updated documentation

### CODE REVIEW

File: sprout/core/edit_resource_properties_field.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        value: the value to assign to the field
+
+    Raises:
+        KeyError: if the specified field does not exist in properties
+
+    Returns:
+        the updated properties object
+    """
+    if field not in properties:
+        raise KeyError(f"Field '{field}' does not exist in properties {properties}.")
```
Comment (martonvago) [inline sprout/core/edit_resource_properties_field.py:18]: I thought this was appropriate, seeing that the nonexistent field is a missing key on the dictionary

Comment (lwjohnst86) [reply sprout/core/edit_resource_properties_field.py:18]: Neat!!

**Review by lwjohnst86 — APPROVED** [review]
Very nice 🤩 Seriously great 😁

**Review by signekb — APPROVED** [review]
Looks good to me ⭐ 🌟

**Pull request merged**